### PR TITLE
Fix: Allows setting adaptive content as other statuses (fixes #1)

### LINKED ISSUE #1: Add option(s) to set content as other status.
Much content we create comes along with a legal requirement to read it. We often get asked that the content isn't removed from the course altogether, but instead available and completed.

### PR DESCRIPTION
**Pull request by joe-replin**
Allows a setting to either mark all adaptive content as complete, optional or unavailable based upon diagnostic assessment results.

### CODE REVIEW

**Comment by joe-replin** [comment]
Fixes #1

File: js/adapt-adaptiveContent.js [hunk 1]
```diff
@@ -175,6 +181,54 @@ define([
...
+  }
+
+  function setAsComplete(ids, saveChanges = false) {
+    if (!ids || ids.length === 0) return;
+    console.log('setAsComplete', ids, saveChanges);
+    ids.forEach(id => {
+      var model = Adapt.findById(id);
+      if (!model) return;
+
+      model.setOnChildren('_isLocked', false);
```
Comment (oliverfoster) [inline js/adapt-adaptiveContent.js:222]: setOnChildren is think, takes object syntax
```js
model.setOnChildren({
  _isLocked: false,
  _isComplete: true,
  _isInteractionComplete: true
});
```
Should probably add `_isInteractionComplete` as well.

File: js/adapt-adaptiveContent.js [hunk 2]
```diff
@@ -159,7 +162,10 @@ define([
     }
 
     const finalAssessmentPageID = finalAssessment.getState().pageId;
-    setAsUnavailable([finalAssessmentPageID], saveChanges);
+
+    if (Adapt.course.get('_adaptiveContent')._setPageStatusAs === "unavailable") {
+      setAsUnavailable([finalAssessmentPageID], saveChanges);
+    };
```
Comment (oliverfoster) [inline js/adapt-adaptiveContent.js:168]: Would calling `checkContentStatusSetting` here do the same thing?

`setAsUnavailable` is the only one of those three functions that is referenced outside of `checkContentStatusSetting`

You could probably condense to something like:
```js
function checkContentStatusSetting(ids, boolean) {
    if (!ids || ids.length === 0) return;
    const statusOption = Adapt.course.get('_adaptiveContent')._setPageStatusAs;
    const logName = `setAs${statusOption[0].toUpperCase()}${statusOption.substring(1)}`;
    console.log(logName, ids, saveChanges);
    ids.forEach(id => {
      var model = Adapt.findById(id);
      if (!model) return;
      switch (statusOption) {
        case 'unavailable':
          model.setOnChildren('_isAvailable', false);
          break;
        case 'optional':
          model.setOnChildren('_isOptional', true);
          model.set('_classes', model.get('_classes') + " diag-optional");
          break;
        case 'complete':
          model.setOnChildren({
            _isLocked: false,
            _isComplete: true,
            _isInteractionComplete: true
          });
          model.set('_classes', model.get('_classes') + " diag-complete");
          break;
      }
    });
    if (saveChanges) saveToOfflineStorage(ids);
  } 
```

**Review by oliverfoster — CHANGES_REQUESTED** [review]
I notice you're using `"` instead of `'` which probably means you've not got eslint configured correctly.

File: js/adapt-adaptiveContent.js (hunk 2 above)
Comment (joe-replin) [reply js/adapt-adaptiveContent.js]: Great suggestion. I've updated based upon your recommendation and tested every switch case locally in my project.

File: js/adapt-adaptiveContent.js (hunk 1 above)
Comment (joe-replin) [reply js/adapt-adaptiveContent.js]: Great to know. Thanks.

**Review by oliverfoster — APPROVED** [review]

File: js/adapt-adaptiveContent.js [hunk 3]
```diff
@@ -159,7 +161,10 @@ define([
     }
 
     const finalAssessmentPageID = finalAssessment.getState().pageId;
-    setAsUnavailable([finalAssessmentPageID], saveChanges);
+
+    if (Adapt.course.get('_adaptiveContent')._setPageStatusAs === 'unavailable') {
+      checkContentStatusSetting([finalAssessmentPageID], saveChanges);
```
Comment (chris-steele) [inline js/adapt-adaptiveContent.js:166]: Content will only be made unavailable if that is explicitly requested via config. Just check that is intended? Obviously if you drop this updated code into an existing project the functionality would change if you forgot to amend the config.

**Comment by joe-replin** [comment]
This has been rebuild as https://github.com/cgkineo/adapt-diagnostic.

**Pull request closed without merging**